token/program-2022/src/serialization.rs

```diff
@@ -1,4 +1,22 @@
-//! serialization module
+//! serialization module - contains helpers for serde types from other crates, deserialization visitors
+
+use {
+    base64::{Engine, prelude::BASE64_STANDARD},
+    serde::de::Error
+};
+
+/// helper function to convert base64 encoded string into a bytes array
+fn base64_to_bytes<const N: usize, E: Error>(v: &str) -> Result<[u8; N], E> {
+    let bytes = BASE64_STANDARD.decode(v).map_err(Error::custom)?;
+
+    if bytes.len() != N {
+        return Err(Error::custom(format!("Length of base64 decoded bytes is not {}", N)));
+    }
+
+    let mut array = [0; N];
+    array.copy_from_slice(&bytes[0..N]);
```

Comment:
nit: How about using `decode_slice` on the pre-allocated array rather than getting a vector and then writing it to an array? https://docs.rs/base64/latest/base64/engine/trait.Engine.html#method.decode_slice

Reply:
Ah I see what you mean, your solution makes the most sense. Thanks for the explanation!